app/controllers/students_controller.rb

```diff
@@ -96,9 +78,9 @@ def service
       :provided_by_educator_name,
       :estimated_end_date
     ])
+    authorized_or_raise! { Student.find(clean_params[:student_id]) }
```

Comment:
This is a real bug - it adds a check that the educator is authorized to read/write data for this student.